AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/MutationSync/OutgoingMutationQueue/ProcessMutationErrorFromCloudOperation.swift

```diff
@@ -311,12 +314,27 @@ class ProcessMutationErrorFromCloudOperation: AsynchronousOperation {
         if case .failure(let error) = cloudResult {
             dataStoreConfiguration.errorHandler(error)
         }
-
-        if case .success(let response) = cloudResult,
-            case .failure(let error) = response {
-            dataStoreConfiguration.errorHandler(error)
+        
+        if case let .success(graphQLResponse) = cloudResult {
+            if case .failure(let error) = graphQLResponse {
+                dataStoreConfiguration.errorHandler(error)
+            } else if case let .success(graphQLResult) = graphQLResponse {
+                guard let reconciliationQueue = reconciliationQueue else {
+                    let dataStoreError = DataStoreError.configuration(
+                        "reconciliationQueue is unexpectedly nil",
+                        """
+                        The reference to reconciliationQueue has been released while an ongoing mutation was being processed.
+                        \(AmplifyErrorMessages.reportBugToAWS())
+                        """
+                    )
+                    finish(result: .failure(dataStoreError))
+                    return
+                }
+                
+                reconciliationQueue.offer([graphQLResult], modelName: mutationEvent.modelName)
+            }
         }
-
+        
         finish(result: .success(nil))
```

Comment:
Yes success and failure here is fairly loose contract between the OutgoingMutationQueue and this operation. I believe even if it's marked as a failure, OutgoingMutationQueue will continue to move on to the next mutation event in the queue (by deleting the mutation event and picking up the next). The important logic in this operation before it finishes is that it's handling the errors. When DataStore can no longer take action on the error, it will call  the `errorHandler` (see line 315 and 320) to inform the client that the error has occurred, before we remove the mutation event.